### DISCUSSION TITLE
Expo secure store adapter warning from Google OAuth

### DISCUSSION THREAD

**Discussion by MorningLit** [supabase/supabase#14306] [Questions]
Hi all, I have implemented Google OAuth for my React Native project and I encountered a warning from Expo.
 **WARN  Provided value to SecureStore is larger than 2048 bytes. An attempt to store such a value will throw an error in SDK 35.**

 Googling this warning led me to this page https://docs.expo.dev/versions/latest/sdk/securestore/ where it also states that 
 **"Size limit for a value is 2048 bytes. An attempt to store larger values may fail. Currently, we print a warning when the limit is reached, however, in a future SDK version an error might be thrown."**
 
 Currently, my implementation of the auth storage is following this guide https://supabase.com/docs/guides/getting-started/tutorials/with-expo 
 and the value that is stored from Google OAuth is a stringified version of Google's OAuth session data object which is around 2800~ characters.
 
 My question is should I be concerned about this warning? Or just ignore it? Thanks in advance

**Comment by MorningLit** [comment]
Additional context on how I implement the OAuth in my code,
referenced and slightly modified from https://github.com/orgs/supabase/discussions/1717#discussioncomment-4174557
```
const returnUrl = Linking.createURL("");
const signInParms = {
    provider: "google" as Provider,
    options: {
        redirectTo: returnUrl,
    },
};
const { data } = await supabase.auth.signInWithOAuth(signInParms);
const authUrl = data.url;
const response = await startAsync({ authUrl, returnUrl });
if (response.type == "success") {
    await supabase.auth.setSession({
        access_token: response.params["access_token"],
        refresh_token: response.params["refresh_token"],
});
```
Updated code due to issues with startAsync:
```
const returnUrl = Linking.createURL("");
const signInParms = {
  provider: "google" as Provider,
  options: {
    redirectTo: returnUrl,
  },
};
const { data } = await supabase.auth.signInWithOAuth(signInParms);
const authUrl = data.url;
const response = await WebBrowser.openAuthSessionAsync(
  authUrl,
  returnUrl
);
if (response.type == "success") {
  var regex = /[?#&]([^=#]+)=([^&#]*)/g,
    params: any = {},
    match;
  while ((match = regex.exec(response.url))) {
    params[match[1]] = match[2];
  }
  await supabase.auth.setSession({
    access_token: params["access_token"],
    refresh_token: params["refresh_token"],
  });
```

**Comment by uoon-dev** [comment]
Same here. I'm not sure how to fix it, but my guess is that when you call the createClient function, you're passing an adapter like ExpoSecureStoreAdapter in the auth.storage field. Like this.
```
import 'react-native-url-polyfill/auto'
import * as SecureStore from 'expo-secure-store'
import { createClient } from '@supabase/supabase-js'

const ExpoSecureStoreAdapter = {
  getItem: (key: string) => {
    return SecureStore.getItemAsync(key)
  },
  setItem: (key: string, value: string) => {
    SecureStore.setItemAsync(key, value)
  },
  removeItem: (key: string) => {
    SecureStore.deleteItemAsync(key)
  },
}

const supabaseUrl = YOUR_REACT_NATIVE_SUPABASE_URL
const supabaseAnonKey = YOUR_REACT_NATIVE_SUPABASE_ANON_KEY

export const supabase = createClient(supabaseUrl, supabaseAnonKey, {
  auth: {
    storage: ExpoSecureStoreAdapter as any,
    autoRefreshToken: true,
    persistSession: true,
    detectSessionInUrl: false,
  },
})
```
I think the issue is caused by the supabase client automatically assigning authentication data to the ExpoSecureStoreAdapter when you log in with Google. All we want to do is assign the provider_token value. We need to fix it so it doesn't automatically setItemAsync the data.
